File: factory.go
package main

import (
	"factory"
)

func main() {
	lining_shoes_factory := factory.LiNingFactory{}
	lining_shoe := lining_shoes_factory.Product("plastic")
	lining_shoe.Wear() // a lining plastic shoe, it's price is 120元;

	adidas_shoe_factory := factory.AdidasFactory{}
	adidas_shoe := adidas_shoe_factory.Product("sports")
	adidas_shoe.Wear() // a adidas sports shoe, it's price is 70元;
}

File: java/src/pattern/Factory.java
package pattern;

import pattern.factory.*;

public class Factory {
    public static void main(String[] args) {
        SimpleFactory simpleFactory = new SimpleFactory();
        Shoe shoeA = simpleFactory.product("sports", "adidas");
        System.out.println(shoeA.getDetail());

        AbstractFactory liningShoeFactory = new AbstractFactory("lining");
        Shoe shoeB = liningShoeFactory.product("cotton");
        System.out.println(shoeB.getDetail());
    }
}
